test/basic/ecs_web_create/ecs_web_create_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/aws/aws-sdk-go/service/ecs"
+	terratest_aws "github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+
+	cac "github.com/aptible/cloud-api-clients/clients/go"
+	"github.com/aptible/terraform-provider-aptible-iaas/internal/client"
+)
+
+func checkSetup() {
+	_, dnsAccountSet := os.LookupEnv("DNS_AWS_ACCOUNT_ID")
+	if !dnsAccountSet {
+		fmt.Printf("DNS_AWS_ACCOUNT_ID environment variable not set\n")
+		os.Exit(1)
+	}
+}
+
+func cleanupAndAssert(t *testing.T, terraformOptions *terraform.Options) {
+	terraform.Destroy(t, terraformOptions)
+
+	// test / assert all failures here
+}
+
+func getAptibleAndAWSVPCs(t *testing.T, ctx context.Context, client client.CloudClient, vpcId, vpcName string) (*cac.AssetOutput, []*terratest_aws.Vpc, error) {
+	vpcAsset, err := client.DescribeAsset(
+		ctx,
+		os.Getenv("ORGANIZATION_ID"),
+		os.Getenv("ENVIRONMENT_ID"),
+		vpcId,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	vpcAws, err := terratest_aws.GetVpcsE(t, []*legacy_aws_sdk_ec2.Filter{
+		{
+			Name:   aws.String("tag:Name"),
+			Values: []*string{aws.String(vpcName)},
+		},
+	}, "us-east-1")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return vpcAsset, vpcAws, nil
+}
+
+func getAptibleAndAWSECSServiceAndCluster(t *testing.T, ctx context.Context, client client.CloudClient, ecsComputeId, ecsClusterName, ecsServiceName string) (*cac.AssetOutput, *ecs.Cluster, *ecs.Service, error) {
+	ecsComputeAsset, err := client.DescribeAsset(
+		ctx,
+		os.Getenv("ORGANIZATION_ID"),
+		os.Getenv("ENVIRONMENT_ID"),
+		ecsComputeId,
+	)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ecsClusterAws := terratest_aws.GetEcsCluster(t, "us-east-1", ecsClusterName)
+	ecsServiceAws := terratest_aws.GetEcsService(t, "us-east-1", ecsClusterName, ecsServiceName)
+
+	return ecsComputeAsset, ecsClusterAws, ecsServiceAws, nil
+}
+
+func insecureHttpClient() *http.Client {
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: transport}
+	return client
+}
+
+func TestECSComputeCreate(t *testing.T) {
+	checkSetup()
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: ".",
+
+		Vars: map[string]interface{}{
+			"organization_id":   os.Getenv("ORGANIZATION_ID"),
+			"environment_id":    os.Getenv("ENVIRONMENT_ID"),
+			"aptible_host":      os.Getenv("APTIBLE_HOST"),
+			"dns_account_id":    os.Getenv("DNS_AWS_ACCOUNT_ID"),
+			"ecs_name":          "ecs-web-test",
+			"container_command": []string{"nginx", "-g", "daemon off;"},
+			"container_image":   "nginx",
+			"container_port":    80,
+			"container_name":    "nginx",
+			"is_public":         true,
+			"is_ecr_image":      false,
+			"vpc_name":          "testecs-web-vpc",
+			"domain":            "aptible-cloud-staging.com",
+			"subdomain":         "test-ecs-integration",
```

Comment:
the only other var i see missing you might want to test (maybe in update command) is : 

```go
ContainerRegistrySecretArn types.String   `tfsdk:"container_registry_secret_arn"`
```

Reply:
I think that's a "flavor" of deployment that needs it's own tf and such. We can add that later when doing the "flavors" but for now I clarified which path this is